### PR TITLE
Lecture 7

### PR DESCRIPTION
**Pull request by Liam-Scott-Russell**

### CODE REVIEW

File: Lecture-7.ipynb [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    "\n",
+    "### Comparison\n",
+    "\n",
+    "| Topic     | Array List Behaviour              | Array Behaviour           |\n",
+    "| :---      | :---                              | :---                      |\n",
+    "| Capacity  | Adjusted as needed                | Fixed at initialization   |\n",
+    "| Printing  | Prints the elements               | Prints the memory address |\n",
+    "| Sizing    | `ArrList.size()`                  | `Arr.length`              |\n",
+    "| Setting   | `ArrList.set(<index>, <value>)`   | `Arr[<index>] = <value>`  |\n",
+    "| Getting   | `ArrList.get(<index>)             | `Arr[<index>]`            |\n",
```
Comment (ncostar) [inline Lecture-7.ipynb]: this block needs a closing backtick

Comment (ncostar) [inline Lecture-7.ipynb]: this block needs a closing backtick

File: Lecture-7.ipynb [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    "* Size is dynamic, rather than fixed\n",
+    "* Can store elements of different types\n",
+    "    * This is because `Object` is a superclass of classes like `Integer` or `String`\n",
+    "\n",
+    "### Disadvantages\n",
+    "\n",
+    "* Array lists are less efficient than arrays\n",
+    "* Array lists lack the `[x]` syntax\n",
+    "    * Must instead use `arrList.get(<index>)` and `arrList.set(<index>, <value>)`\n",
+    "* The base types of array lists must be object, not primitive types\n",
```
Comment (ncostar) [inline Lecture-7.ipynb]: Minor: Might be nicer to refer to them as Reference types rather than Object types, to be consistent with the lecture content?

Comment (Liam-Scott-Russell) [reply Lecture-7.ipynb]: Yup good idea, changed that

**Pull request merged**